source/kraken/apply_disruption.cpp

```diff
@@ -175,6 +176,14 @@ struct add_impacts_visitor : public apply_impacts_visitor {
                 std::move(stoptimes),
                 pt_data);
             LOG4CPLUS_TRACE(log, "New vj has been created " << vj->uri);
+            // Use the corresponding base stop_time for boarding and alighting duration
+            for(auto st: vj->stop_time_list) {
+                const auto base_st = get_base_stop_time(&st);
+                if(base_st) {
+                    st.boarding_time = st.departure_time - base_st->get_boarding_duration();
+                    st.alighting_time = st.arrival_time + base_st->get_alighting_duration();
+                }
+            }
```

Comment:
Yes, definitely needs a test. I completely missed it, I thought I had it but I only tested disruption on line_sections.